helpers/integration_test.go

```diff
@@ -0,0 +1,229 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("Pipeline Adapter", Ordered, func() {
+	var (
+		testpipelineRun           *tektonv1beta1.PipelineRun
+		testpipelineRunwithNoTask *tektonv1beta1.PipelineRun
+		hasApp                    *applicationapiv1alpha1.Application
+		hasSnapshot               *applicationapiv1alpha1.ApplicationSnapshot
+		logger                    logr.Logger
+		sample_image              string
+	)
+
+	BeforeAll(func() {
+
+		hasApp = &applicationapiv1alpha1.Application{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "application-sample",
+				Namespace: "default",
+			},
+			Spec: applicationapiv1alpha1.ApplicationSpec{
+				DisplayName: "application-sample",
+				Description: "This is an example application",
+			},
+		}
+		Expect(k8sClient.Create(ctx, hasApp)).Should(Succeed())
+	})
+
+	BeforeEach(func() {
+		sample_image = "quay.io/redhat-appstudio/sample-image"
+
+		hasSnapshot = &applicationapiv1alpha1.ApplicationSnapshot{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "snapshot-sample",
+				Namespace: "default",
+				Labels: map[string]string{
+					gitops.ApplicationSnapshotTypeLabel:      "component",
+					gitops.ApplicationSnapshotComponentLabel: "component-sample",
+				},
+			},
+			Spec: applicationapiv1alpha1.ApplicationSnapshotSpec{
+				Application: hasApp.Name,
+				Components: []applicationapiv1alpha1.ApplicationSnapshotComponent{
+					{
+						Name:           "component-sample",
+						ContainerImage: sample_image,
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, hasSnapshot)).Should(Succeed())
+		testpipelineRun = &tektonv1beta1.PipelineRun{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pipelinerun-sample",
+				Namespace: "default",
+				Labels: map[string]string{
+					"pipelines.appstudio.openshift.io/type":  "build",
+					"pipelines.openshift.io/used-by":         "build-cloud",
+					"pipelines.openshift.io/runtime":         "nodejs",
+					"pipelines.openshift.io/strategy":        "s2i",
+					"build.appstudio.openshift.io/component": "component-sample",
+				},
+				Annotations: map[string]string{
+					"appstudio.redhat.com/updateComponentOnSuccess": "false",
+				},
+			},
+			Spec: tektonv1beta1.PipelineRunSpec{
+				PipelineRef: &tektonv1beta1.PipelineRef{
+					Name:   "build-pipeline-pass",
+					Bundle: "quay.io/kpavic/test-bundle:build-pipeline-pass",
+				},
+				Params: []tektonv1beta1.Param{
+					{
+						Name: "output-image",
+						Value: tektonv1beta1.ArrayOrString{
+							Type:      "string",
+							StringVal: "quay.io/redhat-appstudio/sample-image",
+						},
+					},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, testpipelineRun)).Should(Succeed())
+
+		testpipelineRun.Status = tektonv1beta1.PipelineRunStatus{
+			PipelineRunStatusFields: tektonv1beta1.PipelineRunStatusFields{
+				TaskRuns: map[string]*tektonv1beta1.PipelineRunTaskRunStatus{
```

Comment:
The integration test pipeline run is not supposed to contain the `build-container` task with the `IMAGE_DIGEST` Tekton result. It should have a number of tasks that each contains the `HACBS_TEST_OUTPUT` Tekton result with JSON contents like in [these examples](https://docs.google.com/document/d/1FJX7TH5wBTcWdU2VYuecmkY5O69DGpItN52A-JaoCtU/edit#heading=h.ffpp5iwdprkv).
In order to cover the different test cases, we'll need to have multiple pipelineRun examples which contain different combinations of tasks that have the `result` field set to `SUCCESS`, `FAILURE` or `SKIPPED`.

Reply:
Done

Reply:
Thanks!